org.eclipse.gef/src/org/eclipse/gef/tools/MarqueeSelectionTool.java

```diff
@@ -236,8 +232,8 @@ public MarqueeSelectionTool() {
 	@Override
 	protected void applyProperty(Object key, Object value) {
 		if (PROPERTY_MARQUEE_BEHAVIOR.equals(key)) {
-			if (value instanceof Integer) {
-				setMarqueeBehavior(((Integer) value).intValue());
+			if (value instanceof Integer intVal) {
+				setMarqueeBehavior(intVal.intValue());
```

Comment:
I can only say that auto unboxing is marked by sonar. So we changed that. Not sure if it was me or @pcdavid .

Reply:
> I can only say that auto unboxing is marked by sonar

Then perhaps it is Sonar who is wrong here... Auto-boxing should only be flagged if it could lead to a potential NullPointerException. Otherwise the rule is flawed.

Reply:
> Auto-unboxing should only be flagged if it could lead to a potential NullPointerException.

Which should still be flagged when one explicitly calls intValue()...

Reply:
Just checked again. I think I had this wrong in my head. Changed it to autounboxing and everything is fine.